platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java

```diff
@@ -110,26 +102,6 @@
  * </ul>
  */
 abstract class AbstractClassGenerator implements ClassGenerator {
-    /**
-     * Types that are allowed to be instantiated directly by Gradle when exposed as a getter on a type.
-     *
-     * @implNote Keep in sync with platforms/documentation/docs/src/docs/userguide/authoring-builds/gradle-properties/properties_providers.adoc
```

Comment:
Regardless of the solution we choose to remove this hard-coded list, we should find a way to keep the connection with documentation. Not sure what form it can best take, maybe an ArchTest or something like that.

Reply:
Ironically, the list in the documentation is not up-to-date 🤔

Reply:
For now I've linked this page in the `ManagedType` javadoc